### DISCUSSION TITLE
[@types/node] fails to find type declarations with typescript v4.4.2

### DISCUSSION THREAD

**Discussion by mondo192** [DefinitelyTyped/DefinitelyTyped#55447] [Issues with a @types package]
When creating a new yarn project with typescript v4.4.2 and adding @types/node to package.json. Typescript fails to find the types even though they appear to be installed. Im running vs code as my text editor and yarn berry. I wonder if this is an issue with @types/node or typescript itself. It works with typescript v4.3.5 but I get this import bug with typescript v4.4.2

```
Cannot find module 'http' or its corresponding type declarations.ts(2307)
```

Also intellisense does not work.

Node: v14.17.5
Yarn: 3.0.1
VSCode: 1.59.1 3866c3553be8b268c8a7f8c0482c0c0177aa8bfa x64
npm: 7.21.1

```
{
  "name": "recipe-api",
  "packageManager": "yarn@3.0.1",
  "private": true,
  "devDependencies": {
    "@types/node": "^16.7.4",
    "typescript": "^4.4.2"
  },
  "dependencies": {
    "fastify": "^3.20.2"
  }
}
```

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
>     "@types/node": "^16.7.4",
The v16 (current, non versioned in `@types/node`) has feature dependency on `node:console`, `node:...` imports.
this depends on runtime support. Can you downgrade types to meet Node version?

**Reply by peterblazejewicz** [reply]
is the `declarations.ts` fastity one?

**Reply by mondo192** [reply]
Its the default http node module. I haven't added any code for fastify other than add the dependency to package.json and the default tsconfig.json via yarn tsc --init.

**Reply by peterblazejewicz** [reply]
Can you please try to run this?
```bash
yarn tsc --traceResolution
```
there should be information how your local setup tries to resolve modules for TSC,
e.g.:
`Module 'http' was resolved as locally declared ambient module in file '[...]/node_modules/@types/node/http.d.ts'.`

**Reply by mondo192** [reply]
I created a gist to show the output. You can find output of ```yarn tsc --moduleResolution``` [here](https://gist.github.com/mondo192/50430c4785cb3d6324d37a1495d95328)

**Reply by mondo192** [reply]
It looks to me like its searching in ```node_modules``` which is not a part of yarn v3.0.1. Yarn berry uses ```./yarn/cache``` as an equivalent to  ```node_modules```  when installing dependencies. I'm curious to what the change was between typescript v4.3.5 and typescript v4.4.2 that seemed to break this? As this "bug" wasn't present in previous minor version v4.3.5

**Reply by peterblazejewicz** [reply]
Can you create new project from scratch, it looks like version of Yarn you're using  (v3) is not resolving dependencies at all (tsc resovles project files properly, but none of external packages are found in node modules). I've used Yarn 1.* here

**Reply by mondo192** [reply]
Sure, I'll write out the steps I did (might be useful for debugging) referenced from yarn 3 official docs [here](https://yarnpkg.com/getting-started/recipes)

[REPO OF PROJECT](https://github.com/mondo192/tester)

```
mkdir tester && cd tester
yarn init -2 -pw
yarn add -D typescript
yarn dlx @yarnpkg/sdks vscode 
yarn plugin import typescript
yarn add -D @types/node
yarn workspace backend init
yarn workspace backend add express
yarn tsc --init
```
Then in the backend workspace I create a server.ts under src and accept the prompt that comes up about which version of typescript to use, I select use workspace sdk version:
```
import http from "http"; <-- this fails due to type declarations missing
```
Project structure:
```
├── package.json
├── packages
│   └── backend
│       ├── package.json
│       ├── README.md
│       └── src
│           └── server.ts
├── README.md
├── tsconfig.json
└── yarn.lock

```

Then running ```yarn tsc --traceResolution``` from the root of the project results in:
```
// omitted the noisy parts
======== Module name 'http' was not resolved. ========
packages/backend/src/server.ts:1:18 - error TS2307: Cannot find module 'http' or its corresponding type declarations.

1 import http from "http";
                   ~~~~~~


Found 1 error.
```
